dd-java-agent/instrumentation/armeria/armeria-grpc-0.84/src/main/java/datadog/trace/instrumentation/armeria/grpc/client/GrpcInjectAdapter.java

```diff
@@ -11,6 +11,9 @@ public final class GrpcInjectAdapter implements CarrierSetter<Metadata> {
 
   @Override
   public void set(final Metadata carrier, final String key, final String value) {
-    carrier.put(Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER), value);
+    Metadata.Key<String> metadataKey = Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER);
+    if (!key.startsWith("ot-baggage-") || !carrier.containsKey(metadataKey)) {
+      carrier.put(metadataKey, value);
+    }
```

Comment:
Why implement a specific logic for baggage? `CarrierSetter` must always replace existing key/value:
```suggestion
    Metadata.Key<String> metadataKey = Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER);
    if (carrier.containsKey(metadataKey)) {
      carrier.removeAll(metadataKey);
    }
```
I don't call always `removeAll` as contains seems more efficient implementation-wise so it is used as a guard for performance.

Reply:
Hmm true 👍. But then is there is there a reason to replace the existing key instead of keeping it? Is that the current expected behavior? (Asking because I'm not sure)

Reply:
So far, the `CarrierSetter` always "set" their values and never cared about any existing prior value.
The main reason is there are carriers that can be queried. So you can expect the carrier setter to "set value if nothing set already".

Reply:
Handled offline. While we could (as-in the `Metadata` type supports reading of data) check if key exists before setting value, it would result in inconsistent behavior with the other carriers used for instrumentations. Thus, we will override the values of all existing keys.